examples/run/cuda/seq_example_cuda.cpp

```diff
@@ -36,15 +41,51 @@
 #include <exception>
 #include <iomanip>
 #include <iostream>
+#include <memory>
 
 namespace po = boost::program_options;
 
 int seq_run(const traccc::full_tracking_input_config& i_cfg,
             const traccc::common_options& common_opts,
             const traccc::detector_input_options& det_opts, bool run_cpu) {
 
-    // Read the surface transforms
-    auto surface_transforms = traccc::io::read_geometry(det_opts.detector_file);
+    // Memory resource used by the application.
+    vecmem::host_memory_resource host_mr;
+
+    // Construct a traccc::geometry object.
+    traccc::geometry surface_transforms;
+    using barcode_map_type = std::map<std::uint64_t, detray::geometry::barcode>;
+    std::unique_ptr<barcode_map_type> barcode_map;
+    if (det_opts.use_detray_detector) {
```

Comment:
Should the following code go in a function somewhere?

Reply:
🤔 Yes, it should...

I'll anyway need to work on this PR a bit after #525 goes in, since that will create a conflict with this. At that point I'll remove the code duplication... 🤔

Reply:
I ended up changing `traccc::io::read_geometry(...)`. 🤔 Please have a look. I'm a bit on the fence about this setup myself, so I'm curious about your opinion.